### ISSUE TITLE
Reorganize `SPARTAN/Software` file system structure

### ISSUE THREAD

**Issue by harrisonCassar** [Bruin-Spacecraft-Group/SPARTAN#12]
Currently, the `SPARTAN/Software` directory's organization is a bit of a mess. Let's reorganize it to be more intuitive. 

A few ideas (strong suggestions, but this likely is not everything): 
- [x] We have a `spartan` directory which would be logical to include only code for the SPARTAN payload system, but we seem to have also the `spartan/src/docs` sub-directory, which contains a few documentation resources we've already made. Let's move this `spartan/docs` directory somewhere else... potentially out to the top-level `Software` directory or as a sister directory to `spartan/src`.
- [x] Similar to the previous idea, let's look at either moving the `notes` directory within this now-moved `docs` directory or just add its files into the `docs` directory itself.
- [x] Within the `spartan` directory, we have the `data` directory. Based on the name, it implies that we have files that our source code works on or with in some way as, well, data... but this is not the case. We instead have source code specifically related to data product and data packet construction. Let's move this source code into something more intuitive, and reserve the `data` directory for code that fits this previously-defined role. 
- [ ] While we're at it, let's figure out what exactly we want the `spartan` directory to hold. Should we have it only hold SPARTAN-specific FSW? Or also source code for the sensor driver library? Or are both too integrated to necessarily separate easily? Should we look to separate this a bit better? If so, let's separate the two (either within `spartan` or by creating another sister directory to `spartan`). This might be a bit more of an arduous task (something worth breaking out into another issue) than moving a few source files, as this possibly involves updating even more relative `#include` paths as well as breaking apart SPARTAN-specific FSW with code that can be generalized into a package-able sensor driver library.
- [x] Since we have been sitting on our newer code organization for a while now, we should consider officially archiving the old source code found in the `old` directory, and potentially looking to remove it from this repository all together. If we're concerned about actually keeping this code, let's move it to a completely different repository, as it currently crowds our repository for not much reason.
- [ ] We have a `main` directory for presumably the `main` source code that uses the SPARTAN sensor driver library that is located within the `spartan` directory. I don't think we necessarily need to have this `main.cpp` essentially alone in a directory of the same name. Let's consider removing the `main` directory all together and moving `main.cpp` into the parent `Software` directory or into the child `spartan` directory.
- [x] Our source code that represents a `mock` implementation to stub out SPARTAN FSW's communication with hardware (located within `spartan/src/mock`) is not technically part of the main SPARTAN FSW, and therefore it intuitively should not be located within this directory. Let's move it somewhere else, possibly to within the top-level `SPARTAN/Software` level (sister directory to `spartan`).

Related to #10 , as we should give a high-level overview of the file system's structure within the README once this is finished.
